### PR TITLE
Scrollable: Convert component to TypeScript

### PR DESCRIPTION
**Pull request by walbo**
## What?
Converts the `Scrollable` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Scrollable` to TypeScript.
- Refactors the tests to TypeScript.
- Refactors the storybook to TypeScript and removed knobs.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Scrollable` continues to function as expected

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@
 -   `CustomGradientPicker`: Refactor away from Lodash ([#41901](https://github.com/WordPress/gutenberg/pull/41901/)).
 -   `SegmentedControl`: Refactor away from `_.values()` ([#41905](https://github.com/WordPress/gutenberg/pull/41905/)).
 -   `DimensionControl`: Refactor docs away from `_.partialRight()` ([#41909](https://github.com/WordPress/gutenberg/pull/41909/)).
+-   `Scrollable`: Convert to TypeScript ([#42016](https://github.com/WordPress/gutenberg/pull/42016/)).
```
Comment (ciampo) [inline packages/components/CHANGELOG.md]: We may need to rebase and add this CHANGELOG entry to the new `Unreleased` section, since version `19.14.0` was released yesterday

File: packages/components/src/scrollable/stories/index.tsx [hunk 2]
```diff
@@ -74,10 +65,8 @@ export const _default = () => {
 					ref={ targetRef }
 					style={ {
 						position: 'absolute',
-						bottom:
-							otherProps.scrollDirection === 'x' ? 'initial' : 0,
-						right:
-							otherProps.scrollDirection === 'y' ? 'initial' : 0,
+						bottom: args.scrollDirection === 'x' ? 'initial' : 0,
+						right: 0,
```
Comment (ciampo) [inline packages/components/src/scrollable/stories/index.tsx:69]: Is this changes on purpose? Previously, the `right` attribute could also have a value of `initial`

**Review by ciampo — APPROVED** [review]
Left a couple of comments, but otherwise changes LGTM 🚀 

Feel free to merge once the comments are addressed

File: packages/components/CHANGELOG.md (hunk 1 above)
Comment (walbo) [reply packages/components/CHANGELOG.md]: Will update 👍

File: packages/components/src/scrollable/stories/index.tsx (hunk 2 above)
Comment (walbo) [reply packages/components/src/scrollable/stories/index.tsx:69]: Yes the change was intended. In my testing the out of view element was outside the scrollable area. Setting `right: 0` made `scrollDirection` work with all both `x`, `y` and `auto`.
